framework/assets/yii.js

```diff
@@ -284,67 +274,185 @@ window.yii = (function ($) {
-                params = {},
-                pair,
-                i;
+                params = {};
 
-            for (i = 0; i < pairs.length; i++) {
-                pair = pairs[i].split('=');
+            for (var i = 0, len = pairs.length; i < len; i++) {
+                var pair = pairs[i].split('=');
                 var name = decodeURIComponent(pair[0].replace(/\+/g, '%20'));
                 var value = decodeURIComponent(pair[1].replace(/\+/g, '%20'));
-                if (name.length) {
-                    if (params[name] !== undefined) {
-                        if (!$.isArray(params[name])) {
-                            params[name] = [params[name]];
-                        }
-                        params[name].push(value || '');
-                    } else {
-                        params[name] = value || '';
+                if (!name.length) {
+                    continue;
+                }
+                if (params[name] === undefined) {
+                    params[name] = value || '';
+                } else {
+                    if (!$.isArray(params[name])) {
+                        params[name] = [params[name]];
                     }
+                    params[name].push(value || '');
                 }
             }
             return params;
         },
 
         initModule: function (module) {
-            if (module.isActive === undefined || module.isActive) {
-                if ($.isFunction(module.init)) {
-                    module.init();
-                }
-                $.each(module, function () {
-                    if ($.isPlainObject(this)) {
-                        pub.initModule(this);
-                    }
-                });
+            if (module.isActive !== undefined && !module.isActive) {
+                return;
             }
+            if ($.isFunction(module.init)) {
+                module.init();
+            }
+            $.each(module, function () {
+                if ($.isPlainObject(this)) {
+                    pub.initModule(this);
+                }
+            });
         },
 
         init: function () {
             initCsrfHandler();
             initRedirectHandler();
-            initScriptFilter();
+            initAssetFilters();
             initDataMethods();
+        },
+
+        // Utility functions
+
+        /**
+         * Returns the URL of the current page without params and trailing slash. Separated and made public for testing.
+         * @returns {string}
+         */
+        getBaseCurrentUrl: function () {
+            return window.location.protocol + '//' + window.location.host;
+        },
+
+        /**
+         * Returns the URL of the current page. Used for testing, you can always call `window.location.href` manually
+         * instead.
+         * @returns {string}
+         */
+        getCurrentUrl: function () {
+            return window.location.href;
         }
     };
 
+    function initCsrfHandler() {
```

Comment:
- Reordered private functions to match the order of according calls in `init` method.
- Moved private utility functions to the end.